### PR TITLE
doc: gas profiles

### PR DESCRIPTION
**Pull request by jakmeier**

### CODE REVIEW

File: docs/architecture/gas/gas_profile.md [hunk 1]
```diff
@@ -1,9 +1,149 @@
...
 
-The transaction runtime charges gas in various places around the code. The
-charges end up as summaries inside an `ActionResult`. More specifically, the
-`gas_burnt` and `gas_used` counters track the total gas required and the
-`profile` field keeps track of what the gas was spent on.
+What if you want to understand the exact gas spending of a smart contract call?
+It would be very complicated to predict exactly how much gas executing a piece
+of WASM code will require, including all host function calls and actions. An
+easier approach is to just run the code on testnet and see how much gas it
+burns. Gas profiles allow to dig deeper and understand the breakdown of the gas
```
Comment (akhi3030) [inline docs/architecture/gas/gas_profile.md:7]:
```suggestion
burns. Gas profiles allow one to dig deeper and understand the breakdown of the gas
```

File: docs/architecture/gas/gas_profile.md [hunk 2]
```diff
@@ -1,9 +1,149 @@
...
+
+The `tx-status` query returns one gas profile for each receipt. The output above
+contains a single gas profile because the transaction only spawned one receipt.
+If there was a chain of cross contract calls, there would be multiple profiles.
+
+Besides receipts, also note the `transaction_outcome` in the output. It contains
+the gas cost for converting the transaction into a receipt. To calculate the
+full gas cost, add up the transaction cost with all receipt costs.
+
+The transaction outcome does currently not have a gas profile, it only shows the
```
Comment (akhi3030) [inline docs/architecture/gas/gas_profile.md:111]:
```suggestion
The transaction outcome currently does not have a gas profile, it only shows the
```

File: docs/architecture/gas/gas_profile.md [hunk 3]
```diff
@@ -1,9 +1,149 @@
...
+total gas spent on converting the transaction. Arguably, it is not necessary to
```
Comment (akhi3030) [inline docs/architecture/gas/gas_profile.md:112]:
```suggestion
total gas spent converting the transaction. Arguably, it is not necessary to
```

File: docs/architecture/gas/gas_profile.md [hunk 4]
```diff
@@ -1,9 +1,149 @@
...
+provide the gas profile since the costs only depend on the list of actions. With
+sufficient understanding of the protocol, one could reverse-engineer the exact
+breakdown simply by looking at the actin list. But adding the profile would
```
Comment (akhi3030) [inline docs/architecture/gas/gas_profile.md:115]:
```suggestion
breakdown simply by looking at the action list. But adding the profile would
```

File: docs/architecture/gas/gas_profile.md [hunk 5]
```diff
@@ -1,9 +1,149 @@
...
+still make sense to make it easier to understand.
+
+## Gas Profile Versions
+
+Depending on the `metadata` version, you should expect a different format of the
```
Comment (akhi3030) [inline docs/architecture/gas/gas_profile.md:120]: should we put a link to where to look up the metadata version?

**Review by akhi3030 — APPROVED** [review]
LGTM

File: docs/architecture/gas/gas_profile.md [hunk 6]
```diff
@@ -1,9 +1,149 @@
...
+gas profile. Version 1 has no profile data at all. Version 2 has a detailed
+profile but some parameters are conflated, so you cannot extract the exact gas
+spending in some cases. Version 3 will have the cost exactly per parameter.
+
+Which version of the profile an RPC node returns depends on the version it had
+when it first processed the transaction. The profiles are stored in the database
+with one version and never updated. Therefore, older transactions will usually
+only have old profiles. However, one could replay the chain from genesis with a
+new nearcore client and generate the newest profile for all transaction in this
```
Comment (akashin) [inline docs/architecture/gas/gas_profile.md:129]: typo: "profile for all transaction**s**"

**Review by akashin — APPROVED** [review]
The explanation makes sense to me! Left a few minor style/typo comments.

File: docs/architecture/gas/gas_profile.md (hunk 5 above)
Comment (jakmeier) [reply docs/architecture/gas/gas_profile.md:120]: It's a field in the output, I wouldn't know what link to put here. Maybe rephrasing it like this makes it clearer?
```
Depending on the version in `receipts_outcome.outcome.metadata.version`, you
should expect a different format of the gas profile.
```

**Comment by jakmeier** [comment]
Thanks @akashin and @akhi3030 for proof reading, appreciate your inputs! I have applied all your suggestions and will add the merge label now.

File: docs/architecture/gas/gas_profile.md [hunk 7]
```diff
@@ -1,9 +1,149 @@
...
+easier approach is to just run the code on testnet and see how much gas it
+burns. Gas profiles allow to dig deeper and understand the breakdown of the gas
+costs per parameter.
+
+**Gas profiles are not very reliable**, in that they are often incomplete and the
+details of how they are computed can change without a protocol version bump.
+
+## Example Transaction Gas Profile
+
+[NEAR CLI](https://docs.near.org/tools/near-cli) can query gas profiles of
```
[the remark this answers is not in the document]
Comment (jakmeier) [reply docs/architecture/gas/gas_profile.md]: yeah that way around it sounds better, thanks

**Pull request merged**